### DISCUSSION TITLE
Count the number of combinations the algorithm tries

### DISCUSSION THREAD

**Discussion by ayushi-ddt** [ahmedfgad/GeneticAlgorithmPython#131] [Q&A] [answered]
Is there a way to count the number of combinations that the algorithm is trying when finding the optimal solution?

**Comment by ahmedfgad** [comment, marked as answer]
@ayushi-ddt, you can set `save_solutions=True` in the constructor of the `pygad.GA` class. This saves every individual solution the algorithm explored into the instance attribute `solutions`. After the algorithm finishes, you can count the number of solutions there:
```python
ga_instance = pygad.GA(..., save_solutions=True)
print(ga_instance.solutions.shape())
```
